src/main/webapp/dev/js/common/checkBrowserVersion.js

```diff
@@ -1,99 +1,28 @@
-// Browser Compatibility and support
-const MSIE = 'Microsoft Internet Explorer';
-const MSIE_LOWEST_VERSION = 9;
-const CHROME = 'Chrome';
-const CHROME_LOWEST_VERSION = 15;
-const FIREFOX = 'Firefox';
-const FIREFOX_LOWEST_VERSION = 12;
-const SAFARI = 'Safari';
-const SAFARI_LOWEST_VERSION = 4;
+/* global UAParser:true */
 
 /**
- * Function to check browser version and alert if browser version is lower than supported
- * Adapted from http://www.javascripter.net/faq/browsern.htm
- *
+ * Function to check whether user's browser is in the
+ * list of recommended browsers and issue a warning otherwise
  */
+function checkBrowser() {
+    const uaParser = new UAParser();
+    const browser = uaParser.getBrowser().name;
+    const recommendedBrowsers = ['Chrome', 'Firefox', 'Safari', 'Edge'];
+    const isSupported = recommendedBrowsers.indexOf(browser) !== -1;
```

Comment:
Let's use `Array.prototype.includes` instead. We need to polyfill it manually (let's use `core-js`) for now, and maybe depending on how Babel is configured in the future too.

I wonder if we missed out any other polyfills other than `String.prototype.includes` though. Related: #7036

Reply:
There's a polyfill provided by MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes . We could use it as well ?